src/main/java/edu/njit/jerse/daikonplusplus/App.java

```
@@ -1,11 +1,660 @@
+    final String fullRunCp = JavaRunner.joinCp(selfCp, classesDir.toString(), userClasspath);
+
+    JavaRunner.compileWithAutoFilter(srcRoot, classesDir, fullCompileCp, /*maxPasses*/ 10);
+    final Path runLog = srcRoot.resolve("daikonpp-run.log");
+    JavaRunner.run(mainClass, fullRunCp, programArgs, runLog);
+
+    long exitLines = 0;
+    try {
+      exitLines = Files.lines(runLog).filter(s -> s.contains("\"phase\":\"EXIT\"")).count();
+    } catch (IOException ioe) {
+      // ignore
+    }
+    System.out.println(">>> Run log — EXIT events: " + exitLines);
+
+    // --- Phase 4: parse run log and generate the results
+    final Set<UUID> falsified = LogParser.readFalsifiedIds(runLog);
+    final Set<UUID> executed = LogParser.readExecutedIds(runLog);
+    final Set<UUID> nonCompiled = LogParser.readNonCompiledIds(srcRoot);
+
+    final Map<UUID, edu.njit.jerse.daikonplusplus.App.RecordLite> all =
+        parseRegistryLite(cfg.registryPath());
+
+    // compiled = all − nonCompiled
+    final Set<UUID> compiledIds = new HashSet<>(all.keySet());
+    compiledIds.removeAll(nonCompiled);
+
+    // compute a verdict for every known record and persist outcomes
+    Map<UUID, InvariantRegistry.Outcome> outcomes = new HashMap<>();
+    for (var e : all.entrySet()) {
+      UUID id = e.getKey();
+      boolean compiled = compiledIds.contains(id);
+      boolean exec = executed.contains(id);
+
+      InvariantRegistry.Verdict verdict;
+      if (nonCompiled.contains(id)) {
+        verdict = InvariantRegistry.Verdict.FAILED_TO_COMPILE;
+      } else if (exec && falsified.contains(id)) {
+        verdict = InvariantRegistry.Verdict.FALSIFIED;
+      } else if (exec) {
+        verdict = InvariantRegistry.Verdict.HELD;
+      } else if (compiled) {
+        verdict = InvariantRegistry.Verdict.NEVER_EXECUTED;
+      } else {
+        verdict =
+            InvariantRegistry.Verdict
+                .PROPOSED; // seen in registry but neither compiled nor executed
```

Should this case be possible? If not, it may be better to throw an exception.